what-is-chalk.mdx

```diff
@@ -18,215 +18,467 @@ import { PyDiffEditor } from '@/components/Editor'
 
-Many data sources (like vendor APIs) are too slow for online use cases
-or charge a high dollar cost-per-call. Chalk lets you optimize latency
-and cost by defining declarative caching policies that are well-integrated
-throughout the system. You no longer have to manage data sources such as
-Redis, Memcached, or DynamoDB, or spend time tuning cache-warming pipelines.
+Some data sources, like LLMs or 3rd party APIs, are expensive or slow to call at inference time.
+Chalk supports caching features to optimize for latency and cost:
 
-Add a caching policy with one line of code in your feature definition:
+- define cache windows inline with your features
+- override staleness at query-time when fresh data is critical
+- pre-warm caches and backfill pipelines for performance
+
+Add a caching policy with one line of code in our feature definition:
 
 <PyDiffEditor html={addCachingCode} />
 
-Optionally warm feature caches by executing resolvers on a schedule:
-
-```sql
--- resolves: User
--- source: snowflake
--- cron: 1h
-select
-    user_id as id,
-    credit_report,
-    ts
-from credit_scores
+
+### How do I deploy and query features with Chalk?
+
+After defining pipelines, deploy features to a branch in < 1s with:
+Chalk automatically detects and uses your current git branch name without requiring you to explicitly specify it:
+
+```shell
+chalk apply --branch
 ```
 
-Or override staleness tolerances at query time when you need fresher
-data for your models:
+Query for features against this branch and perform QA workflows on branch deployments to ensure pipelines perform as expected:
 
-```py
-from chalk.client import ChalkClient
+```shell
+chalk query --branch --in user.id=47
+```
+
+This command returns the user's features from your branch deployment, showing all computed fields for the specified `user.id`.
 
-ChalkClient().query(
-    input={
-        User.id: 123,
-    },
-    output=[User.credit_report],
-    max_staleness={User.credit_report: "1m"}
-)
 ```
+Using '--out=user'
+Results
+https://chalk.ai/environments/{environment}/query-runs/{query_id}
+Branch: elvis
+Environment: {environment}
+
+ Name                                             Hit?  Value
+─────────────────────────────────────────────────────────────────────────────────────────────────────
+ user.average_rating_given                  4.3125
+ user.birthday                              "1987-12-28"
+ user.created_at                            "2024-08-07T20:54:42.942294+00:00"
+ user.email                                 "98178ad58fe8481db74996996d6e8de7@google.ru"
+ user.first_name                            "Marseda"
+ user.id                                    47
+ user.last_name                             "Karkaletsis"
+ user.review_count                          32
+ user.total_orders_placed                   48
+ user.unique_products_inquired_about        203
+```
+
+Once validated, promote the deployment to production, which is served at a unique preview URL.
```

Comment:
we don't really do the preview URLs anymore. the blue/green deployments are our answer for rolling this stuff out smoothly

Reply:
done!